rdkit/Chem/AllChem.py

```diff
@@ -451,6 +450,102 @@ def AssignBondOrdersFromTemplate(refmol, mol):
+        just a heuristic that could result in stereoisomers being lost.
+
+      - onlyUnassigned: if set (the default), stereocenters which have specified stereochemistry
+        will not be perturbed
+
+      - verbose: toggles how verbose the output is
+
+    A small example with 3 chiral centers (8 theoretical stereoisomers):
+    >>> from rdkit.Chem import AllChem
+    >>> m = AllChem.MolFromSmiles('OC1OC(C2)(F)C2(Cl)C1')
+    >>> isomers = AllChem.GenerateAllStereoisomers(m)
+    >>> len(isomers)
+    8
+    >>> for smi in sorted(AllChem.MolToSmiles(x,isomericSmiles=True) for x in isomers):
+    ...     print(smi)
+    O[C@@H]1C[C@@]2(Cl)C[C@@]2(F)O1
+    O[C@@H]1C[C@@]2(Cl)C[C@]2(F)O1
+    O[C@@H]1C[C@]2(Cl)C[C@@]2(F)O1
+    O[C@@H]1C[C@]2(Cl)C[C@]2(F)O1
+    O[C@H]1C[C@@]2(Cl)C[C@@]2(F)O1
+    O[C@H]1C[C@@]2(Cl)C[C@]2(F)O1
+    O[C@H]1C[C@]2(Cl)C[C@@]2(F)O1
+    O[C@H]1C[C@]2(Cl)C[C@]2(F)O1
+
+    Because the molecule is constrained, not all of those isomers can
+    actually exist. We can check that:
+    >>> isomers = AllChem.GenerateAllStereoisomers(m,tryEmbedding=True)
+    >>> len(isomers)
+    4
+    >>> for smi in sorted(AllChem.MolToSmiles(x,isomericSmiles=True) for x in isomers):
+    ...     print(smi)
+    O[C@@H]1C[C@@]2(Cl)C[C@@]2(F)O1
+    O[C@@H]1C[C@]2(Cl)C[C@]2(F)O1
+    O[C@H]1C[C@@]2(Cl)C[C@@]2(F)O1
+    O[C@H]1C[C@]2(Cl)C[C@]2(F)O1
+
+    By default the code only expands unspecified stereocenters:
+    >>> m = AllChem.MolFromSmiles('O[C@H]1OC(C2)(F)C2(Cl)C1')
+    >>> isomers = AllChem.GenerateAllStereoisomers(m)
+    >>> len(isomers)
+    4
+    >>> for smi in sorted(AllChem.MolToSmiles(x,isomericSmiles=True) for x in isomers):
+    ...     print(smi)
+    O[C@@H]1C[C@@]2(Cl)C[C@@]2(F)O1
+    O[C@@H]1C[C@@]2(Cl)C[C@]2(F)O1
+    O[C@@H]1C[C@]2(Cl)C[C@@]2(F)O1
+    O[C@@H]1C[C@]2(Cl)C[C@]2(F)O1
+
+    but we can change that behavior:
+    >>> isomers = AllChem.GenerateAllStereoisomers(m, onlyUnassigned=False)
+    >>> len(isomers)
+    8
+
+
+
+    """
+    tm = Mol(m)
+    res = []
+    if onlyUnassigned:
+        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True) if y=='?']
+    else:
+        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True)]
+    nCenters = len(possibleCenters)
+    if not nCenters:
+        return (tm,)
+    bitflag = (1<<nCenters)-1
+    while bitflag>=0:
+        tm = Mol(m)
+        for i in range(nCenters):
+            if bitflag & 1<<i:
+                tm.GetAtomWithIdx(possibleCenters[i]).SetChiralTag(CHI_TETRAHEDRAL_CCW)
+            else:
+                tm.GetAtomWithIdx(possibleCenters[i]).SetChiralTag(CHI_TETRAHEDRAL_CW)
+        if tryEmbedding:
+            ntm = AddHs(tm)
+            cid = EmbedMolecule(ntm,randomSeed=bitflag)
+        else:
+            cid = 1
+        if cid>= 0:
+            res.append(tm)
```

Comment:
If this was intended to be python only, I would suggest using a generator.

Even in C++ it might be useful to have a lazy iterator.